packages/edit-site/src/components/sidebar-navigation-screen-pattern/use-navigation-menu-title.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import { NAVIGATION_POST_TYPE } from '../../utils/constants';
+
+export default function useNavigationMenuTitle( id ) {
+	return useSelect(
+		( select ) => {
+			if ( ! id ) {
+				return undefined;
+			}
+
+			const editedRecord = select( coreStore ).getEditedEntityRecord(
+				'postType',
+				NAVIGATION_POST_TYPE,
+				id
+			);
+
+			// Do not display a 'trashed' navigation menu.
+			return editedRecord.status === 'trash'
+				? undefined
+				: editedRecord.title;
+		},
```

Comment:
The hook is called `useNavigationMenuTitle` so I expect it to return the title and not return "empty" even if the record is "trashed". It feels like the logic to show or not something that is "trashed" doesn't really belong to a `useNavigationMenuTitle` hook. And we may just want to filter some request or something.

Anyway, not that important since it's just internal API but I thought I'd mention.

Reply:
This started as inline `useSelect`; conditions are more "visible" in that case, and I don't see any issue returning `undefined` based on record status.

I guess we can return `{ title, viewable }` from the hook to make it more explicit.

Reply:
To be honest it's not important and I'm fine if leave it but I can clarify my reasoning:

- `useNavigationMenuTitle` nothing in the title of the hook suggests anything about the status of the navigation menu. I also think a trashed navigation menu should keep its title.
- Returning `{ title, viewable }`: yeah that's better but than the name of the hook is not adequate anymore. It feels like this needs two hooks or just leave inline useSelect (what's wrong with that :P).

I know I'm getting into weeds, so don't waste time on this.

Reply:
I always appreciate your feedback, @youknowriad!